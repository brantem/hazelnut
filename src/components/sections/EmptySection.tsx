import Button from 'components/Button';

type EmptySectionProps = {
  title: string;
  action: {
    children: React.ReactNode;
    onClick: () => void;
  };
};

const EmptySection = ({ title, action }: EmptySectionProps) => {
  return (
    <section className="flex h-full flex-col items-center justify-center dark:text-white" data-testid="empty-section">
      <p className="mb-3" data-testid="empty-section-title">
        {title}
      </p>

      <Button data-testid="empty-section-action" onClick={action.onClick} className="dark:bg-white dark:text-black">
        {action.children}
      </Button>
    </section>
  );
};

export default EmptySection;
