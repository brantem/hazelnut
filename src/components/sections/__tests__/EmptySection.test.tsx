import { render, act, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import EmptySection from 'components/sections/EmptySection';

test('EmptySection', async () => {
  const onClick = vi.fn();
  const { container } = render(<EmptySection title="Title" action={{ children: 'Action', onClick }} />);

  expect(container).toMatchSnapshot();
  act(() => screen.getByText('Action').click());
  expect(onClick).toHaveBeenCalled();
});
