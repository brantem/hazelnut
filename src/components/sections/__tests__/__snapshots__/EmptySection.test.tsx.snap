// Vitest Snapshot v1

exports[`EmptySection 1`] = `
<div>
  <section
    class="flex h-full flex-col items-center justify-center dark:text-white"
    data-testid="empty-section"
  >
    <p
      class="mb-3"
      data-testid="empty-section-title"
    >
      Title
    </p>
    <button
      class="rounded-full focus:outline-none focus:ring-2 focus:ring-neutral-500 dark:focus:ring-neutral-700 disabled:opacity-70 bg-black text-white px-4 py-1.5 dark:bg-white dark:text-black"
      data-testid="empty-section-action"
    >
      Action
    </button>
  </section>
</div>
`;
